src/Routes/Devices/AddDeviceModal.js

```diff
@@ -98,33 +99,46 @@ const AddDeviceModal = ({
     apiWithToast(dispatch, addDevicesToGroupFunc, statusMessages);
   };
   return (
-    <Modal
-      isOpen={isModalOpen}
-      closeModal={() => setIsModalOpen(false)}
-      title={`Add to ${
-        inventoryGroupsEnabled && useWorkspacesRename ? 'workspace' : 'group'
-      }`}
-      submitLabel="Add"
-      additionalMappers={{
-        'search-input': {
-          component: SearchInputApi,
-        },
-        'create-group-btn': {
-          component: CreateGroupButton,
-          closeModal: () => {
-            setIsCreateGroupModalOpen(true);
-            setIsModalOpen(false);
+    <>
+      <Modal
+        isOpen={isModalOpen && !isCreateGroupModalOpen}
+        closeModal={() => setIsModalOpen(false)}
+        title={`Add to ${
+          inventoryGroupsEnabled && useWorkspacesRename ? 'workspace' : 'group'
+        }`}
+        submitLabel="Add"
+        additionalMappers={{
+          'search-input': {
+            component: SearchInputApi,
+          },
+          'create-group-btn': {
+            component: CreateGroupButton,
+            closeModal: () => {
+              setIsCreateGroupModalOpen(true);
```

Comment:
Just a question, I'm just not sure here:
Why don't we call `setIsModalOpen(false)` here anymore?

Reply:
it causes the entire component to be removed from the DOM (`src/Routes/Devices/Inventory.js`, line 365) and since the child modal is exactly that (a child modal), it needs the host modal to be in the DOM

Reply:
Gotcha! :)